torchtitan/components/checkpoint.py

```diff
@@ -641,6 +647,7 @@ def _ft_save(self, step: int) -> None:
         self.save_future = self.dcp_save(
             self.ft_states, checkpoint_id=checkpoint_id, async_mode=AsyncMode.ASYNC
         )
+        self._purge_stale_ft_checkpoints()
```

Comment:
do we need to call this here? I thought the purge thread do the deletion

Reply:
Yes, but the directories to purge need to be added to the queue.